spec/models/bag_man_request_spec.rb

```diff
@@ -59,31 +59,42 @@
     end
 
     context "when not cancelled" do
+      let(:replication) {
+        mock = double(:replication_transfer)
+        allow(mock).to receive(:cancel!)
+        allow(mock).to receive(:from_node).and_return FromNode.new("fake_namespace")
+        mock
+      }
+      let(:request) {
+        request = Fabricate(:bag_man_request)
+        allow(request).to receive(:replication_transfer).and_return(replication)
+        request
+      }
       before(:each) do
-        @repl_mock = double(:replication_transfer)
-        allow(@repl_mock).to receive(:cancel!)
-        allow(@repl_mock).to receive(:from_node).and_return FromNode.new("fake_namespace")
-        @request = Fabricate(:bag_man_request)
-        allow(@request).to receive(:replication_transfer).and_return(@repl_mock)
         allow(Client::Repl::CancelJob).to receive(:perform_later)
       end
       it "sets the cancel_reason" do
-        @request.cancel!('testing')
-        expect(@request.reload.cancel_reason).to eql('testing')
+        request.cancel!('testing')
+        expect(request.reload.cancel_reason).to eql('testing')
       end
       it "calls replication_transfer.cancel!" do
-        expect(@repl_mock).to receive(:cancel!)
-        @request.cancel!('testing')
+        expect(replication).to receive(:cancel!)
+        request.cancel!('testing')
       end
       it "passes the reason to replication_transfer.cancel!" do
-        expect(@repl_mock).to receive(:cancel!).with('testing')
-        @request.cancel!('testing')
+        request.cancel!('testing')
+        expect(replication).to have_received(:cancel!).with('testing', anything)
+      end
+      it "passes last_error as the detail to replication_transfer.cancel!" do
+        request.update!(last_error: "some_error")
+        request.cancel!('testing')
+        expect(replication).to have_received(:cancel!).with(anything, request.last_error)
       end
       it "enqueues a Client::Repl::CancelJob" do
         expect(Client::Repl::CancelJob).to receive(:perform_later)
-          .with(@repl_mock, "fake_namespace", "replicate",
+          .with(replication, "fake_namespace", "replicate",
             "update_replication", "ReplicationTransferAdapter")
-        @request.cancel!('testing')
+        request.cancel!('testing')
       end
```

Comment:
What happens when `Client::Repl::CancelJob#perform_later` raises an exception?  Does it cancel the transaction that cancels the replication_transfer?  If it does, should there be a `retry` block somewhere?